x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -0,0 +1,600 @@
+# Installation and Upgrade of Prebuilt Rules
+
+This is a test plan for the workflows of installing and upgrading prebuilt rules.
+
+Status: `in progress`. The current test plan matches `Milestone 2` of the [Rule Immutability/Customization](https://github.com/elastic/security-team/issues/1974) epic. It does not cover any past functionality that was removed or functionality to be implemented in the future. The plan is about to change in the future Milestones.
+
+## Useful information
+
+### Tickets
+
+- [Rule Immutability/Customization](https://github.com/elastic/security-team/issues/1974) epic
+- [Ensure full test coverage for existing workflows of installing and upgrading prebuilt rules](https://github.com/elastic/kibana/issues/148176)
+- [Write test plan and add test coverage for the new workflows of installing and upgrading prebuilt rules](https://github.com/elastic/kibana/issues/148192)
+- [Document the new UI for installing and upgrading prebuilt detection rules](https://github.com/elastic/security-docs/issues/3496)
+
+### Terminology
+
+- **EPR**: [Elastic Package Registry](https://github.com/elastic/package-registry), service that hosts our **Package**.
+
+- **Package**: `security_detection_engine` Fleet package that we use to distribute prebuilt detection rules in the form of `security-rule` assets (saved objects).
+
+- **Real package**: actual latest stable package distributed and pulled from EPR via Fleet.
+
+- **Mock rules**: `security-rule` assets that are indexed into the `.kibana_security_solution` index directly in the test setup, either by using the ES client _in integration tests_ or by an API request _in Cypress tests_.
+
+- **Air-gapped environment**: an environment where Kibana doesn't have access to the internet. In general, EPR is not available in such environments, except the cases when the user runs a custom EPR inside the environment.
+
+- **CTA**: "call to action", usually a button, a link, or a callout message with a button, etc, that invites the user to do some action.
+  - CTA to install prebuilt rules - at this moment, it's a link button with a counter (implemented) and a callout with a link button (not yet implemented) on the Rule Management page.
+  - CTA to upgrade prebuilt rules - at this moment, it's a tab with a counter (implemented) and a callout with a link button (not yet implemented) on the Rule Management page.
+
+### Assumptions
+
+- Below scenarios only apply to prebuilt detection rules.
+- Users should be able to install and upgrade prebuilt rules on the `Basic` license and higher.
+- EPR is available for fetching the package unless explicitly indicated otherwise.
+- Only the latest **stable** package is checked for installation/upgrade and pre-release packages are ignored.
+
+### Non-functional requirements
+
+- Notifications, rule installation and rule upgrade workflows should work:
+  - regardless of the package type: with historical rule versions or without;
```

Comment:
What does Historical versions mean? Previous versions? If so, which ones? We should specify which versions are we using for testing on a test plan, because not all the version have the same relevance. i.e: Most of our current users are on 7.17.x version, and we should add extra effort on testing that the upgrade from this version is successful. Same happens with lower 8.x versions unless there haven't been any major changes on rules from lower 8.x versions until this 8.9.

Reply:
@vgomez-el Great question! This is an architectural aspect. We ship historical (1 latest + N previous) versions of each prebuilt rule in the prebuilt rules package. So it's not about the stack version, and not about the version of the prebuilt rules package itself.

More info about this can be found here:

- https://docs.google.com/document/d/1vDO2aTDLrb3LohxsaxXwnZK4Yk_3_24bTsBBeYNo6k0/edit#heading=h.iu5r7ke72629
- https://github.com/elastic/kibana/issues/137420